Firmware/Marlin_main.cpp

```diff
@@ -3133,11 +3136,13 @@ bool gcode_M45(bool onlyZ, int8_t verbosity_level)
 				bool result = sample_mesh_and_store_reference();
 				if (result)
 				{
-					if (calibration_status() == CALIBRATION_STATUS_Z_CALIBRATION)
-						// Shipped, the nozzle height has been set already. The user can start printing now.
-						calibration_status_store(CALIBRATION_STATUS_CALIBRATED);
-						final_result = true;
-					// babystep_apply();
+                    if (calibration_status() == CALIBRATION_STATUS_Z_CALIBRATION)
+                    {
```

Comment:
And which one seems to be correct? With final_result=true inside the block or not? 
Not considering the previous misleadingly indented state of code...

Reply:
`final_result` needs to be set outside the block as done by this patch. Otherwise, this function would return _failse_ when an _onlyZ_ home is successfully executed.